### ISSUE TITLE
Candidate merging can cause candidacies to be removed

### ISSUE THREAD

**Issue by andylolz** [DemocracyClub/yournextrepresentative#118]
The version history for https://candidates.democracyclub.org.uk/person/2111/stuart-jeffery includes the following merge:
![screen shot 2017-04-23 at 19 40 24](https://cloud.githubusercontent.com/assets/464193/25316370/ed565ffc-285d-11e7-96d7-33cc2af5e59e.png)
You can see that historical candidacies for GE2010 and GE2015 were removed in this merge (which shouldn’t happen!)

Note that this can also cause candidacies for locked posts to be removed (as in the above example, in fact) which is particularly bad (because the list of candidacies for the post has been checked and verified by a person).

**Comment by mhl** [comment]
I haven't managed to reproduce this :( I've made a pull request with tests that attempt to reproduce it, but they don't, I'm afraid. They might still be useful as more complex test examples, or ones we can tweak later to reproduce the problem, so I think it's probably still worth merging.

The timestamp on this bad merge was 1st May 2016, so it's possible that this was due to a bug that has since been fixed - there were a few commits relating to merging around that time. (In particular, one was to make sure that the merging happened in a transaction, but I'm not sure I can attribute that to a failure midway through merging since I can't find a corresponding error email.) I can't say I'm very happy with just saying "maybe it's been fixed since?", but I think given the high priority of some of the other issues we have it's probably best to put this aside for the moment and watch carefully for any similar problems with merges from now on.
